File: commands/fun/yomomma.go
package fun

import (
  "github.com/sapphire-cord/sapphire"
  "net/http"
  "io/ioutil"
  "encoding/json"
)

// Get a random yo momma joke.
// Usage: [@user]
// Aliases: urmom
func Yomomma(ctx *sapphire.CommandContext) {
  res, err := http.Get("http://api.yomomma.info")

  if err != nil {
    ctx.Reply("Error: %s", err)
    return
  }

  body, err := ioutil.ReadAll(res.Body)

  if err != nil {
    ctx.Reply("Error: %s", err)
    return
  }

  var j struct {
    Joke string `json:"joke"`
  }

  err = json.Unmarshal(body, &j)

  if err != nil {
    ctx.Reply("Error: %s", err)
    return
  }

  if ctx.HasArgs() {
    ctx.Reply(ctx.Arg(0).AsUser().Mention() + ", " + j.Joke)
  } else {
    ctx.Reply(j.Joke)
  }
}

File: commands/fun/yomomma.js
const Command = require("../../structures/Command.js");
const fetch = require("node-fetch");

class YoMomma extends Command {
  constructor(...args) {
    super(...args, {
      description: "Yo momma so fat.",
      aliases: ["urmom"],
      cooldown: 3,
      usage: "yomomma [@user]"
    });
  }
  
  async run(ctx, [user]) {
    user = await this.verifyUser(ctx, user, true);

    const { joke } = await fetch("http://api.yomomma.info")
      .then((res) => res.json());

    return ctx.reply(`${user}, ${joke}`);
  }
}

module.exports = YoMomma;
